hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/operand/OperandCheckerProgram.java

```diff
@@ -46,4 +56,31 @@ public boolean check(HazelcastCallBinding callBinding, boolean throwOnFailure) {
 
         return res;
     }
+
+    private boolean hasAssignment(SqlCall call) {
+        for (SqlNode operand : call.getOperandList()) {
+            if (operand != null && operand.getKind() == SqlKind.ARGUMENT_ASSIGNMENT) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private OperandChecker findCheckerByParamName(SqlCallBinding binding, int index) {
```

Comment:
I couldn't really understand what this method does. The `OperandCheckerProgram` could be viewed as a convenient wrapper that executes predefined checkers for every parameter one by one. With this change, we start doing something completely different when there is a parameter. Could you please clarify a couple of things?
1. What is the purpose of the argument assignment operator? Can it be used in any context, or only in some designated operators, like table functions? 
2. What does `HazelcastFunction.getParamNames()` return? There is not JavaDoc and no implementation in IMDG, so the semantics is not clear.

Reply:
1. An alternative way of calling a function. Instead of positional arguments, one can use named ones, like `name => 'value'`. In theory, every operator could support named parameters.
2. Well, parameter names. It is used by Calcite whenever argument assignment is used.

Reply:
Got it, thanks.